tests/Microsoft.Identity.Test.Unit/PublicApiTests/TelemetryClientTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.IdentityModel.Abstractions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.PublicApiTests
+{
+    [TestClass]
+    public class TelemetryClientTests : TestBase
+    {
+        [TestMethod]
+        public void TelemetryClientExperimental()
+        {
+            ITelemetryClient telemetryClient = new TelemetryClient(TestConstants.ClientId);
+
+            var e = AssertException.Throws<MsalClientException>(() => ConfidentialClientApplicationBuilder
+                .Create(TestConstants.ClientId)
+                .WithClientSecret("secret")
+                .WithTelemetryClient(telemetryClient)
+                .Build());
+
+            Assert.AreEqual(MsalError.ExperimentalFeature, e.ErrorCode);
+        }
+    }
+
+    internal class TelemetryClient : ITelemetryClient
+    {
+        public TelemetryClient(string clientId)
+        {
+            ClientId = clientId;
+        }
+
+        public string ClientId { get; set; }
+
+        public void Initialize()
+        {
+            throw new NotImplementedException();
+        }
+
+        public bool IsEnabled()
```

Comment:
Is there no need to test these other methods in this class? are they already tested somewhere else?

Reply:
The other method implementation will follow with the next task where I am logging the telemetry. For now they are not needed.